mist-core/src/main/java/edu/snu/mist/core/driver/MistDriver.java

```diff
@@ -134,23 +197,40 @@ public void onNext(final StartTime startTime) {
           .setMemory(mistDriverConfigs.getTaskMemSize())
           .setNumberOfCores(mistDriverConfigs.getNumTaskCores())
           .build());
+      requestor.submit(EvaluatorRequest.newBuilder()
+          .setNumber(1)
+          .setMemory(mistDriverConfigs.getMasterMemSize())
+          .setNumberOfCores(mistDriverConfigs.getNumMasterCores())
+          .build());
       LOG.log(Level.INFO, "Requested Evaluator.");
     }
   }
 
   public final class EvaluatorAllocatedHandler implements EventHandler<AllocatedEvaluator> {
     @Override
     public void onNext(final AllocatedEvaluator allocatedEvaluator) {
-      LOG.log(Level.INFO, "Submitting Context to AllocatedEvaluator: {0}", allocatedEvaluator);
-      final String taskId = "MistTask-" + taskIndex.getAndIncrement();
-      final JVMProcess jvmProcess = jvmProcessFactory.newEvaluatorProcess()
-          .setMemory(mistDriverConfigs.getTaskMemSize())
-          .addOption("-XX:NewRatio=" + mistDriverConfigs.getNewRatio())
-          .addOption("-XX:ReservedCodeCacheSize=" + mistDriverConfigs.getReservedCodeCacheSize() + "m");
-      allocatedEvaluator.setProcess(jvmProcess);
-      allocatedEvaluator.submitContext(ContextConfiguration.CONF
-          .set(ContextConfiguration.IDENTIFIER, taskId)
-          .build());
+      final EvaluatorDescriptor descriptor = allocatedEvaluator.getEvaluatorDescriptor();
+      if (descriptor.getMemory() == mistDriverConfigs.getMasterMemSize()
```

Comment:
Is it safe to check whether this allocated evaluator is  `master` or not by using the memory size? What happen if the memory size of master and task is equal?

Reply:
Maybe we can use `requestor.setRuntimeName` ?

Reply:
I have tried `requestor.setRuntimeName()`, but it didn't work. For YARN, it is okay to allocate evaluator only by memory because each container has no limitation on using CPU.